polkadot-protocol/protocol-components/transactions/transactions-weights-and-fees.md

```diff
@@ -0,0 +1,220 @@
+- Burn operations are used to pay for a transaction based on its internal logic. For example, a transaction might burn funds from the sender if the transaction creates new storage items to pay for the increased the state size
+- Limits enable you to enforce constant or configurable limits on specific operations. For example, the default [Staking pallet](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/staking){target=\_blank} only allows nominators to nominate 16 validators to limit the complexity of the validator election process
+
+It is important to note that if you query the chain for a transaction fee, it only returns the inclusion fee.
+
+## Default Weight Annotations
+
+All dispatchable functions in the Polkadot SDK must specify a weight. The way of doing that is using the annotation-based system that lets you combine fixed values for database read/write weight and/or fixed values based on benchmarks. The most basic example would look like this:
+
+```rust
+--8<-- 'code/polkadot-protocol/trasactions/transactions-weights-and-fees/default-weight-annotation.rs'
+```
+
+Note that the [`ExtrinsicBaseWeight`](https://crates.parity.io/frame_support/weights/constants/struct.ExtrinsicBaseWeight.html){target=\_blank} is automatically added to the declared weight to account for the costs of simply including an empty extrinsic into a block.
+
+### Weights and Database Read/Write Operations
+
+To make weight annotations independent of the deployed database backend, they are defined as a constant and then used in the annotations when expressing database accesses performed by the dispatchable:
+
+```rust
+--8<-- 'code/polkadot-protocol/trasactions/transactions-weights-and-fees/weight-read-write-annotation.rs'
+```
+
+This dispatchable allows one database to read and two to write, in addition to other things that add the additional 20,000. Database access is generally every time a value declared inside the [`#[pallet::storage]`](https://paritytech.github.io/polkadot-sdk/master/frame_support/pallet_macros/attr.storage.html){target=\_blank} block is accessed. However, unique accesses are counted because after a value is accessed, it is cached, and reaccessing it does not result in a database operation. That is:
+
+- Multiple reads of the exact value count as one read
+- Multiple writes of the exact value count as one write
+- Multiple reads of the same value, followed by a write to that value, count as one read and one write
+- A write followed by a read-only counts as one write
+
+### Dispatch Classes
+
+Dispatches are broken into three classes:
+
+- Normal
+- Operational
+- Mandatory
+
+If a dispatch is not defined as Operational or Mandatory in the weight annotation, the dispatch is identified as Normal by default. You can specify that the dispatchable uses another class like this:
```

Comment:
```suggestion
If a dispatch is not defined as `Operational` or `Mandatory` in the weight annotation, the dispatch is identified as `Normal` by default. You can specify that the dispatchable uses another class like this:
```

Since this is referring to the class in code